samples/matter/lock/CMakeLists.txt

```diff
@@ -18,7 +18,9 @@ if(CONF_FILE)
     get_filename_component(CONFIG_FILE_NAME ${CONF_FILE} NAME)
 endif()
 
-if(NOT CONFIG_FILE_NAME STREQUAL "prj_no_dfu.conf")
+if(CONFIG_FILE_NAME STREQUAL "prj_thread_wifi_switched.conf")
```

Comment:
Add `AND ${BOARD} STREQUAL "nrf5340dk_nrf5340_cpuapp"`?

Reply:
hmm, I think this wouldn't be sufficient as we would still use `_thread_wifi_switched` config but with _dfu partition layout. Also, we don't seem to verify at the CMake level all valid configurations, such as prj.conf on nRF21. We can introduce such early checks but consistently for all samples.